clip_benchmark/datasets/builder.py

```diff
@@ -158,18 +168,42 @@ def build_dataset(dataset_name, root="root", transform=None, split="test", downl
         ds = voc2007.PASCALVoc2007(root=root, set="train" if train else "test", transform=transform, download=download, **kwargs)
     elif dataset_name == "mscoco_captions":
         # https://github.com/mehdidc/retrieval_annotations/releases/tag/1.0.0(annotations)
-        if split == "train":
-            archive_name = "train2014.zip"
-        elif split in ("val", "test"):
-            archive_name = "val2014.zip"
-        else:
-            raise ValueError(f"split should be train or val or test for `{dataset_name}`")
-        root_split = os.path.join(root, archive_name.replace(".zip", ""))
-        if not os.path.exists(root_split):
-            print(f"Downloading mscoco_captions {archive_name}...")
-            if not os.path.exists(os.path.join(root, archive_name)):
-                call(f"wget http://images.cocodataset.org/zips/{archive_name} --output-document={root}/{archive_name}", shell=True)
-            call(f"unzip {root}/{archive_name} -d {root}", shell=True)
+        def get_archive_name(target_split):
+            if target_split == "train":
+                return "train2014.zip"
+            elif target_split in ("val", "test"):
+                return "val2014.zip"
+            else:
+                raise ValueError(f"split should be train or val or test for `{dataset_name}`")
+
+        def download_mscoco_split(target_split):
+            archive_name = get_archive_name(target_split)
+            root_split = os.path.join(root, archive_name.replace(".zip", ""))
+            if not os.path.exists(root_split):
+                print(f"Downloading mscoco_captions {archive_name}...")
+                if not os.path.exists(os.path.join(root, archive_name)):
+                    call(f"wget http://images.cocodataset.org/zips/{archive_name} --output-document={root}/{archive_name}", shell=True)
+                call(f"unzip {root}/{archive_name} -d {root}", shell=True)
+
+        download_mscoco_split(split)
+        root_split = os.path.join(root, get_archive_name(split).replace(".zip", ""))
+        if (language != 'en'):
+            # Multilingual MS-COCO or XTD-Dataset
+            # 
+            # If we are loading any multilingual version we override the annotation file
+            # We also override the root_split, since the multilingual_en has images from all splits
+            # Instead the image paths have been altered to include to split-folders
+            # We therefore also check so that all splits have been downloaded
+            from clip_benchmark.datasets import multilingual_mscoco
+            annotation_file = os.path.join(root, multilingual_mscoco.IMAGE_INDEX_FILE)
```

Comment:
it feels like coco and multilingual coco are 2 different datasets
can we simply consider them as such ?

Reply:
Very good point! I have now implemented it as its own dataset.